### DISCUSSION TITLE
Where do guys put the complex but low reusability components?

### DISCUSSION THREAD

**Discussion by s3xysteak** [nuxt/nuxt#28715] [General]
In SPA, I often place complex but low reusable components next to routing components:

- views
  - user.vue
  - Detail.vue
  - Config.vue

But in nuxt, doing so under pages will result in a new route being created. Did everyone simply put it in the components folder? This doesn't quite satisfy me because it looks like a universal component. Or ss there any way to set blocking rules for routing? I tried `ignore`, but this will make me lost the HMR.

**Comment by masyoudi** [comment]
are you looking for [this](https://deltener.com/blog/nuxt-enterprise-patterns-component-management/)?
